### DISCUSSION TITLE
Cylinder with lamellar and statistical morpho

### DISCUSSION THREAD

**Discussion by FeygaLiu** [neperfepx/neper#945] [Q&A] [answered]
Hello every,

I am looking for a way to create a cylinder that can be cut into three layers (10,103,10 mm from bottom to top), with the middle 103 mm height containing the weighted normal statistical morphology. 

#415 #558 #626 seems similar; but I am not sure how to create in a cylinder geometry.

Thank you!

**Comment by FeygaLiu** [comment]
I have tried some code based on the previous discussion, but it does not work. 

neper -T -n "3::msfile(n_msfile)" -domain "cylinder(10,10,30)" -morpho "lamellar(w=2:5:3,pos=start,v=(0,0,1))::msfile(morpho_msfile)" -o d2

**n_msfile**
1 1
2 from_morpho
3 1

**morpho_msfile**
1	voronoi
2	diameq:lognormal(2,0.3),1-sphericity:lognormal(0.145,0.03)
3	voronoi

**Comment by rquey** [comment, marked as answer]
Your ms files are fine. Update your Neper (to version 4.10.2-3) and use this command:
```
neper -T -n "from_morpho::msfile(n_msfile)" -domain "cylinder(123,10,30)" -morpho "lamellar(w=10:103,pos=start,v=(0,0,1))::msfile(morpho_msfile)" -o d2
```
